packages/dapper/src/queryString.ts

```diff
@@ -0,0 +1,125 @@
+import { isNumberArray, isStringArray } from "./typeguards";
+
+type QueryStringable = string | number | boolean | string[] | number[];
+
+interface QsItem<T extends QueryStringable> {
+  key: string;
+  value?: T;
+  impotent?: T;
+}
+
+interface DefinedQsItem<T extends QueryStringable> extends QsItem<T> {
```

Comment:
Minor note, but I feel like `RequiredQsItem` would be a more standard naming scheme.

Reply:
I don't think it quite fits in this case? The interface is used when we've checked that the given value is not undefined. It's still not required as in a "required field", we've just filtered out those items where the value was not explicitly defined.

Reply:
Ah, that makes more sense. I got confused thinking from type hinting perspective, i.e. if I were to type a query param interface of which some properties are required and some optional. I think this way around the naming logic is sound 👍